### ISSUE TITLE
Create abstract implementations of configuration-dependent and configuration-independent property pages

### ISSUE THREAD

**Issue by tmeschter** [dotnet/project-system#5777]
Our current property pages contain either configuration-dependent or configuration-independent properties. For the new App Designer we should define one or more abstract page types that handle the Configuration and Platform drop-down menus, including retrieving the available and current configuration and platform and exposing them to derived types.

Dependent on #5761, #5762, and #5776.

**Comment by tmeschter** [comment]
Re-assigning to @drewnoakes and @melytc as this concerns the new property page UI.

**Comment by drewnoakes** [comment]
Closing in favour of #6627.
